### DISCUSSION TITLE
Dashy Upgrade from 3.0.1 to 3.1.1 via "update" command

### DISCUSSION THREAD

**Discussion by strangedesign13** [tteck/Proxmox#3604] [Help Needed]
When trying to use the script to update from 3.0.1 to 3.1.1. 

✓ Stopped Dashy
 ✓ Backed up conf.yml
 / Updating Dashy to 3.1.1   
[ERROR] in line 80: exit code 0: while executing command npm run build &> /dev/null

Running on Debian GNU/Linux 12.

**Comment by tteck** [comment]
Try increasing resources. Also make sure you have enough storage.

**Comment by strangedesign13** [comment]
I'm already setup with the 2 CPU, 2GB of RAM and 6 GB of Disk (all per the script setup). Any proposed changes to those resources? I should not there is nothing us on this LXC but dashy and webmin.

**Comment by strangedesign13** [comment]
I increase the resources to 4 CPU, 4GB RAM, 1GB Swap and 12 GB of Boot.
![Screenshot 2024-08-28 at 11 23 50](https://github.com/user-attachments/assets/3dffb8cd-2abc-4a87-bcc3-efff948f280f)

Still the same issue, so doubtful it's a resource issue. Any other ideas?

**Comment by CrazyWolf13** [comment]
@strangedesign13 
Without further info this is really hard to diagnose. Can you try first downloading the script via wget and then manually removing the following: ` &> /dev/null ` from `npm run build &> /dev/null` then run run the downloaded script, and show the output here.

**Reply by strangedesign13** [reply]
Here is the output:

> NODE_OPTIONS=--openssl-legacy-provider vue-cli-service build

node: --openssl-legacy-provider is not allowed in NODE_OPTIONS

[ERROR] in line 80: exit code 0: while executing command npm run build

**Reply by tteck** [reply]
Where is `NODE_OPTIONS=` coming from?

**Reply by strangedesign13** [reply]
My guess is it is in node.js. Is there a specific version of node.js the script is expecting?

**Reply by tteck** [reply]
Are you executing the command `update` in the LXC console? The `NODE_OPTIONS=--openssl-legacy-provider vue-cli-service build/NODE_OPTIONS=yarn vue-cli-service build` was in an old script. https://github.com/tteck/Proxmox/commit/37f42468ba295b42f4cd636fb31d8eacbd2703ab

**Reply by strangedesign13** [reply]
Yes. I have run `update` from within the LXC console, used the `bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/dashy.sh)"` from https://tteck.github.io/Proxmox/
downloaded the script from [https://github.com/tteck/Proxmox/raw/main/ct/dashy.sh](https://github.com/tteck/Proxmox/raw/main/ct/dashy.sh) and ran it locally.

Same result from both with `[ERROR] in line 80: exit code 0: while executing command npm run build &> /dev/null`

I got the `node: --openssl-legacy-provider is not allowed in NODE_OPTIONS` error after removing the `&> /dev/null` from `npm run build &> /dev/null` when I downloaded the script locatlly as @CrazyWolf13 suggested.

**Reply by CrazyWolf13** [reply]
@strangedesign13 
Can you first try showing what the current node options are by using `export`.

And then remove the current options like suggested here in the second-top answer.

https://stackoverflow.com/questions/72866798/error-message-node-openssl-legacy-provider-is-not-allowed-in-node-options

Remember you have to run it through the LXC console or even better directly via ssh, running through `pct enter xxx` will result in a fail.

**Reply by strangedesign13** [reply]
No NODE_OPTIONS even set:

```
declare -x HOME="/root"
declare -x HUSHLOGIN="FALSE"
declare -x INVOCATION_ID="REDACTED"
declare -x LANG="C"
declare -x LOGNAME="root"
declare -x MAIL="/var/mail/root"
declare -x MOTD_SHOWN="pam"
declare -x OLDPWD
declare -x PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
declare -x PWD="/root"
declare -x SHELL="/bin/bash"
declare -x SHLVL="1"
declare -x SYSTEMD_EXEC_PID="161"
declare -x TERM="xterm-256color"
declare -x USER="root"
```

**Reply by BytemasterGB** [reply]
Any solution in sight? I have the same problem as OP running script in LXC Console or using update.

**Reply by BytemasterGB** [reply]
BTW. Thanks for all your work.

**Reply by CrazyWolf13** [reply]
I'd suggest copying the dashy config file and setting up a new LXC vis ttecks script.
That's probably the best way

**Reply by tteck** [reply]
I just made the update verbose to help see the error
